### PR TITLE
Added new Pull Request Code Analysis task and associated files.

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: Tasks/PullRequestCodeAnalysis/typings.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "Agent.Tasks.PRCA",
+    "dependencies": {},
+    "globalDependencies": {
+        "chai": "registry:dt/chai#3.4.0+20160317120654",
```
Comment (bgavrilMS) [inline Tasks/PullRequestCodeAnalysis/typings.json]: Not sure we need chai and mocha anymore

File: Tasks/PullRequestCodeAnalysis/task.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+    "id": "683B88A0-96C3-11E6-8625-95B67567393D",
+    "name": "PullRequestCodeAnalysis",
+    "friendlyName": "Pull Request Code Analysis",
+    "description": "Post code analysis issues as comments on a pull request",
+    "helpMarkDown": "[More Information](   NEEDS LINK    )",
```
Comment (bgavrilMS) [inline Tasks/PullRequestCodeAnalysis/task.json]: Log a task for this so that we don't forget.

File: Tasks/PullRequestCodeAnalysis/task.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "minimumAgentVersion": "1.99.0",
+    "instanceNameFormat": "Post comments to pull request",
+    "inputs": [
+        {
+            "name": "messageLimit",
+            "type": "string",
+            "label": "Comment limit",
+            "defaultValue": "100",
+            "required": true,
+            "helpMarkDown": "The maximum number of comments that can be posted. Larger values may take longer to complete."
```
Comment (bgavrilMS) [inline Tasks/PullRequestCodeAnalysis/task.json]: How about - "Posting too many comments makes the PR hard to read. High severity issues are posted first"

File: Tasks/PullRequestCodeAnalysis/task.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        "Build"
+    ],
+    "author": "Microsoft Corporation",
+    "demands": [],
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "minimumAgentVersion": "1.99.0",
```
Comment (bgavrilMS) [inline Tasks/PullRequestCodeAnalysis/task.json]: Not sure we need this.

File: Tasks/PullRequestCodeAnalysis/package.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    "url": "git+ssh://git@github.com/Microsoft/vsts-tasks.git"
+  },
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/Microsoft/vsts-tasks/issues"
+  },
+  "dependencies": {
+    "fs": "0.0.2",
+    "q": "1.4.1",
+    "vso-node-api": "^5.0.5",
```
Comment (bgavrilMS) [inline Tasks/PullRequestCodeAnalysis/package.json]: You should take a depdency specifically vso-node-api: 5.1.1

File: Tasks/PullRequestCodeAnalysis/task.json (hunk 2 above)
Comment (HSAR) [inline Tasks/PullRequestCodeAnalysis/task.json]: Done, thanks for the reminder.

File: Tasks/PullRequestCodeAnalysis/task.json (hunk 3 above)
Comment (HSAR) [inline Tasks/PullRequestCodeAnalysis/task.json]: Good points. Changed.

File: Tasks/PullRequestCodeAnalysis/task.json (hunk 4 above)
Comment (HSAR) [inline Tasks/PullRequestCodeAnalysis/task.json]: I'm not sure either, but every task.json has this. It still builds without, though, so let's remove it and see.

**Comment by bgavrilMS** [comment]
LGTM

**Pull request merged**